### DISCUSSION TITLE
Improve Documentation - Proposals

### DISCUSSION THREAD

**Discussion by mellanslag-de** [statelyai/xstate#3704] [General]
As I read the documentation I stumbled across a few things. I hope it's okay that I just list them here in one Issue/Discussion together:

1. Move Documentation of the sendParent action to the section "Actions". Currently it's in 'Invoking Services'. It may make sense there because thats where I need them, but if I already know about them and want to come back to read more, I would first look in the section "Actions" (because well, its an action). We could then only link to the new location from "Invoking Services".
2. "Events -> Null events" kinda says the exact same as "Transitions -> Eventless ("Always") Transitions". With its own (different!) examples and way of phrasing it. I'd merge these paragraphes, and just link from the deprecated syntax to the new one with a short paragraph that compares the syntax difference. Or whats the timeline, will it get removed completely soon anyway?
3. I missed an overview at the beginning of every 'object' (event, action, guard, ...) with how the maximum specification looks like. Like which options can I give, and which parameters does the function get (context, event, state/action/_event). Whats the state on this? https://github.com/statelyai/xstate/issues/3165#issuecomment-1090059979
4. https://xstate.js.org/docs/guides/states.html#state-nextevents Does this function also check guards and exclude those transitions that aren't possible?
5. I'm not too familiar with the exact definition of 'Effect', but doesn't it make sense to at least mention Actors here as well? https://xstate.js.org/docs/guides/effects.html

I'll be happy to help update the docs, just wanted to hear your opinion first 😅

And I noticed some other things but they are more like questions/workarounds, and I think I'll rather post an individial question each, if even only for documentation reasons?

**Comment by Andarist** [comment]
cc @laurakalbag

**Comment by laurakalbag** [comment]
@mellanslag-de thank you so much for your suggestions! We’ve got all-new documentation landing in beta in the next week. I think we address a few of your suggestions, but we’d really appreciate your feedback when they’re out. I can update you here if that works for you?

**Comment by mellanslag-de** [comment]
That sounds great! So yeah, please keep me updated, and I'll be happy to give feedback!

**Reply by laurakalbag** [reply]
@mellanslag-de our updated docs are now live at https://stately.ai/docs. We’ve still got a lot to add, but would really appreciate any feedback you’d like to give us! You can let me know here, we also have a # documentation channel on [our Discord](https://discord.gg/xstate) and a [feature request board especially for documentation](https://statelyai.canny.io/docs).

**Comment by ellingtonjp** [comment]
Another suggestion!

The [reddit tutorial](https://xstate.js.org/docs/tutorials/reddit.html) appears out of date. The last section, [putting it all together](https://xstate.js.org/docs/tutorials/reddit.html#putting-it-all-together), specifies the `useService` hook, but that doesn't exist anymore (at least not that I could find).

I'm new to xstate so not sure how to should be updated. Replacing `useService` with `useActor` worked, so maybe that's all that needs to change?

Side note: as a newcomer, I've found the term "service" to be confusing. It seems to be defined as an [actor created from a machine](https://xstate.js.org/docs/tutorials/reddit.html#putting-it-all-together), an [interpreted running statechart](https://xstate.js.org/docs/guides/interpretation.html), and any of [machine/promise/callback/observable](https://xstate.js.org/docs/guides/communication.html#the-invoke-property) (although the first breakout box [here](https://xstate.js.org/docs/guides/actors.html#actor-api) seems to differentiate services from promises/observables/callbacks). A clear definition (glossary maybe?) would be helpful, IMO.
